libs/imap/Makefile

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (C) 2017 Lucian Cristian <lucian.cristian@gmail.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+# updated to work with latest source from abrasive
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=imap
+PKG_VERSION:=2007f
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=http://ftp.ntua.gr/pub/net/mail/imap
+PKG_HASH:=53e15a2b5c1bc80161d42e9f69792a3fa18332b7b771910131004eb520004a28
+
+PKG_MAINTAINER:=Lucian Cristian <lucian.cristian@gmail.com>
+
+PKG_LICENSE:=GPL-V2
+PKG_LICENSE_FILES:=LICENSE.txt
+
+PKG_FIXUP:=autoreconf
+PKG_BUILD_DEPENDS:= +libpthread +libopenssl
+PKG_BUILD_PARALLEL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/uw-imap
+  SECTION:=libs
+  CATEGORY:=Libraries
+  TITLE:=University of Washington IMAP toolkit
+  URL:=https://www.washington.edu/imap/
+endef
+
+define Package/uw-imap/description
+    The UW IMAP tookit includes the following:
+
+	c-lient library: an API (application programming interface) used to build email clients and servers,
+	including support for IMAP,POP3, SMTP and NNTP protocols and for local mailbox file access on Unix and Windows
+endef
```

Comment:
Please check intendation. Two single whitespace should be enough. Also please mention, that c-client is a for static linking - otherwise people could be confused when looking after the binaries in the target system.

Reply:
Can we modifying it to generate a dynlib instead?

Reply:
didn't try it, I'll test, but I think this should be left only for devel, the only packages that are using this is asterisk, php and prayer and for imap and pop server we already have a very good server

Reply:
I'm not sure that's really the point.  There are other libraries we build dynamically only if they only link to a single app...

Reply:
You have read https://www.washington.edu/imap/IMAP-FAQs/#6.3? So I guess we could package a dynlib for OpenWrt/LEDE, but then we should not expect upstream support anymore. (on the other hand, I don't know whether upstream is still very active - I didn't check). Also a dynlib only makes sense we same lib is shared... here this means that php and asterisk must be installed on the same system... Then both (php and asterisk) must be patched to use the dynlib instead of the static lib... So patches for both projects are required which have to be maintained... So to short-cut: I think it's not worth the efforts to create a dynlib here. Or to make it more explicit: I would NAK a php patch to use the dynlib of imap because I don't see a chance to get such a patch upstream.

Reply:
Yeah, I've read it... but I don't really buy it.  It sounds like a lame excuse to have no ABI discipline.

Reply:
If this is no upstream support, then every time there's a CVE we're going to be on the hook for fixing it.

Reply:
lots of important distros are using this, I think they'll be aware if something happens, didn't look at RedHat sources for patches but I'll verify